src/metadata.ts

```diff
@@ -13,6 +13,104 @@ export interface BaseMetadataEntry {
   metadataType: MetadataType;
 }
 
+export type Milliseconds = number;
+export type Seconds = number;
+
+/**
+ * Time range expressed in milliseconds since Unix epoch.
+ */
+export interface AbsoluteTimeRange {
+  /**
+   * The start date of the range.
+   */
+  start: Milliseconds;
+  /**
+   * The end date of the range.
+   */
+  end?: Milliseconds;
+}
+
+/**
+ * Time range expressed in seconds since playback start.
+ */
+export interface RelativeTimeRange {
+  /**
+   * The start time of the range.
+   */
+  start: Seconds;
+  /**
+   * The end time of the range.
+   */
+  end?: Seconds;
+}
+
+/**
+ * Represents in-playlist timed metadata from an HLS `#EXT-X-DATERANGE` tag.
+ */
+export interface DateRangeMetadataEntry extends BaseMetadataEntry {
+  metadataType: MetadataType.DATERANGE;
+  /**
+   * The unique identifier for the date range.
+   */
+  id: string;
+  /**
+   * The class associated with the date range.
+   */
+  classLabel?: string;
+  /**
+   * Time range of the entry relative to Unix Epoch.
+   *
+   * If the metadata represents an instantaneous event, {@link AbsoluteTimeRange.end} should be equal
+   * to {@link AbsoluteTimeRange.start}.
+   * An omitted {@link AbsoluteTimeRange.end} indicates an open-ended range.
+   *
+   * @platform iOS, tvOS
+   */
+  absoluteTimeRange?: AbsoluteTimeRange;
+  /**
+   * Time range of the entry relative to the beginning of the playback.
+   *
+   * @platform Android
+   */
+  relativeTimeRange?: RelativeTimeRange;
+  /**
+   * The declared duration of the range in seconds.
+   */
+  duration?: Seconds;
+  /**
+   * The planned duration of the range in seconds.
+   *
+   * Used for live streams where the actual end time may not be known yet.
+   *
+   * @platform Android
+   */
+  plannedDuration?: Seconds;
+  /**
+   * Indicates whether the date range ends at the start of the next date range
+   * with the same {@link classLabel}.
+   *
+   * @platform Android
+   */
+  endOnNext?: boolean;
+  /**
+   * The `CUE` attribute values from an `#EXT-X-DATERANGE` tag.
+   *
+   * Empty array if the attribute is not present.
+   * `"PRE"` triggers before playback; `"POST"` triggers after completion; `"ONCE"` limits triggering to once.
+   * When multiple values are provided, the first takes precedence (e.g., `"PRE,POST"` -> `"PRE"`).
+   * `"PRE"` and `"POST"` are re-playable unless `"ONCE"` is included.
+   *
+   * Note: Applies only to HLS Interstitial opportunities (pre-, mid-, post-roll).
```

Comment:
Please use `@remarks` to highlight a block in API docs
```suggestion
   * @remarks Applies only to HLS Interstitial opportunities (pre-, mid-, post-roll).
```